centreon/www/class/centreon-clapi/centreonHost.class.php

```diff
@@ -1010,6 +1013,49 @@ public function setmacro($parameters)
                 )
             );
         }
+        // If the macro is password and a vault configuration exists
+        if ((int) $params[3] === 1 && ($vaultConfiguration = $this->getVaultConfiguration()) !== null) {
+            $logger = $this->getLogger();
+            $httpClient = new \CentreonRestHttp();
+            $clientToken = $this->authenticateToVault($vaultConfiguration, $logger, $httpClient);
+            $resourceEndpoint = $vaultConfiguration->getStorage() . '/monitoring/hosts/' . $hostId;
+            $hostSecrets = $this->getHostSecretsFromVault(
+                $vaultConfiguration,
+                $resourceEndpoint,
+                $clientToken,
+                $logger,
+                $httpClient
+            );
+            $hostSecrets['_HOST' . strtoupper($params[1])] = $params[2];
+            $this->writeSecretsInVault(
+                $vaultConfiguration,
+                $hostId,
+                $clientToken,
+                $hostSecrets,
+                $logger,
+                $httpClient
+            );
```

Comment:
```suggestion
            $logger = $this->getLogger();
            $httpClient = new \CentreonRestHttp();
            $clientToken = $this->authenticateToVault($vaultConfiguration, $logger, $httpClient);
            $resourceEndpoint = $vaultConfiguration->getStorage() . '/monitoring/hosts/' . $hostId;
            $hostSecrets = $this->getHostSecretsFromVault(
                $vaultConfiguration,
                $resourceEndpoint,
                $clientToken,
                $logger,
                $httpClient
            );
            $hostSecrets['_HOST' . strtoupper($params[1])] = $params[2];
            $this->writeSecretsInVault(
                $vaultConfiguration,
                $hostId,
                $clientToken,
                $hostSecrets,
                $logger,
                $httpClient
            );
```
refactor into a method writeMacroSecretsInVault()